quartz/plugins/transformers/citations.ts

```diff
@@ -0,0 +1,33 @@
+import rehypeCitation from "rehype-citation"
+import { QuartzTransformerPlugin } from "../types"
+
+export interface Options {
+  bibliographyFile: string
+  suppressBibliography: boolean
+}
+
+const defaultOptions: Options = {
+  bibliographyFile: "./bibliography.bib",
+  suppressBibliography: false,
+}
+
+export const Citations: QuartzTransformerPlugin<Partial<Options> | undefined> = (userOpts) => {
+  const opts = { ...defaultOptions, ...userOpts }
+  return {
+    name: "Citations",
+    htmlPlugins() {
+      return [
+        [
+          rehypeCitation,
+          {
+            bibliography: opts.bibliographyFile,
+            suppressBibliography: opts.suppressBibliography,
+            // Until popovers are implemented to not show for citation links, I
+            // don't want to link citations
```

Comment:
https://github.com/jackyzha0/quartz/blob/v4/quartz%2Fcomponents%2Fscripts%2Fpopover.inline.ts#L10-L12

Perhaps adding this attribute to the citation links could be a way to fix this?

Reply:
Hi @saberzero1. Thanks for taking the time! This sounds indeed like the way to go. However, I'm not sure where I can add the property to the links generated by `rehype-citation`. So, I'd be very grateful if you could provide some pointers. I'm happy to implement but right now I would not know where to start. Also because `noPopover` does not seem to be used anywhere.

Reply:
`noPopover` is used here:

https://github.com/jackyzha0/quartz/blob/b4236e5142c31829cf809c0fbc8370ac22b6d1ba/quartz/plugins/transformers/gfm.ts#L31-L39

Basically, you want the link element to have a `data-no-popover="true"` property.

For example, from the `rehype-citation` repository:

```markdown
My markdown text [@Nash1950]
```

```html
<div>My markdown text (Nash, 1950)</div>
<div id="refs" class="references csl-bib-body">
  <div class="csl-entry">
    Nash, J. (1950). Equilibrium points in n-person games.
    <i>Proceedings of the National Academy of Sciences</i>, <i>36</i>(1), 48–49.
  </div>
</div>
```

You'd probably want something like this as the resulting html:

```html
<div>My markdown text (Nash, 1950)</div>
<div id="refs" class="references csl-bib-body" data-no-popover="true">
  <div class="csl-entry">
    Nash, J. (1950). Equilibrium points in n-person games.
    <i>Proceedings of the National Academy of Sciences</i>, <i>36</i>(1), 48–49.
  </div>
</div>
```

Reply:
we can add another HTML transform here to add `noPopover` to anything with and id that starts with `citation--` or something

reading https://github.com/timlrx/rehype-citation/blob/4e797f54fbe4083adc7efcbb2125176eb5615a9c/test/note.js#L40

Reply:
> we can add another HTML transform here to add `noPopover` to anything with and id that starts with `citation--` or something
> 
> reading https://github.com/timlrx/rehype-citation/blob/4e797f54fbe4083adc7efcbb2125176eb5615a9c/test/note.js#L40

Is this something we want to do in this pull request, or is this a separate issue? I can imagine having an option to add `noPopover` to HTML transform might be something that could be re-used by other (future) plugins.

Reply:
no this should specifically be for the citation plugin

Reply:
I used `rehype-rewrite` to do this and used `startWith('#bib')` for all `a` tags to single out those that need the `data-no-popover=true`. I tested this locally and it seems to do exactly what we want.